### PR TITLE
Python: More miscellaneous fixes

### PR DESCRIPTION
**Pull request by bje-**
## What does this PR do?

Silences more Pylint warnings.

### CODE REVIEW

File: gdal/swig/python/osgeo/gdal_array.py [hunk 1]
```diff
@@ -263,12 +263,10 @@ def DatasetReadAsArray(ds, xoff=0, yoff=0, win_xsize=None, win_ysize=None, buf_o
         interleave = True
         xdim = 2
         ydim = 1
-        bdim = 0
```
Comment (rouault) [inline gdal/swig/python/osgeo/gdal_array.py]: each time you touch swig/python/osgeo/* files you need to do changes in the sources files in swig/include/

File: autotest/ogr/ogr_osm.py [hunk 2]
```diff
@@ -979,9 +979,7 @@ def ogr_osm_18():
 
     ds = ogr.Open('data/two_points.pbf')
     lyr = ds.GetLayerByName('points')
-    count = 0
```
Comment (rouault) [inline autotest/ogr/ogr_osm.py]: this change should be revert. For performance reason GetFeatureCount() on a OSM layer (which is what is called with len(lyr)) returns -1.

File: autotest/utilities/test_ogr2ogr.py [hunk 3]
```diff
@@ -2721,7 +2721,7 @@ def hexify_double(val):
     return val.rstrip('L').lstrip('0x').zfill(16).upper()
 
 
-def check_identity_transformation(x, y, srid):
+def check_identity_transformation(x, y):
```
[the remark this answers is not in the document]
Comment (bje-) [reply autotest/utilities/test_ogr2ogr.py]: Not sure why Pylint warned that `srid` was unused. Will revert (and double check to see why).

File: autotest/ogr/ogr_osm.py (hunk 2 above)
Comment (bje-) [reply autotest/ogr/ogr_osm.py]: How is the following not semantically identical?

```
-    count = 0
-    for f in lyr:
-        count += 1
+    count = len(lyr)
```

Comment (bje-) [reply autotest/ogr/ogr_osm.py]: That is indeed delicious. :-)

**Pull request merged**